### DISCUSSION TITLE
Can we fetch only a particular column from a collection?

### DISCUSSION THREAD

**Discussion by habibullahturkmen** [pocketbase/pocketbase#2433] [Q&A] [answered]
Can we fetch particular columns from a collection in Pocketbase? fetch only vehicle names from vehicles collection?

**Comment by ganigeorgiev** [comment]
Support to limit the returned API response fields using the `?fields` query parameter is already implemented but currently it is available only in the [`develop` branch](https://github.com/pocketbase/pocketbase/tree/develop).

It will be part of the next v0.16.0 release (there are no ETAs yet).

**Reply by habibullahturkmen** [reply]
Thank you so much for your support.

**Reply by habibullahturkmen** [reply]
How can I use the `?fields` query parameter?

``.getFullList({
        filter: "fields=name"
 });``
 
 the above code didn't work. I'm using the develop branch.

**Reply by ganigeorgiev** [reply, marked as answer]
`fields` is a separate query parameter:

```js
await pb.collection("vehicles").getFullList({
    fields: "id,name"
})
```
The above should return an array with only the `id` and `name` fields of the `vehicles` collection.

**Reply by habibullahturkmen** [reply]
Thank you so much.

**Reply by kisamoto** [reply]
Will this be enforceable via rules or is this only selecting a subset? 

i.e. Could a user manipulate a request (example below) to get access to a secret_col or would/could this be blocked by rules?

```js
await pb.collection("vehicles").getFullList({
    fields: "id,name,secret_col"
})
```

I assume that if we wanted this behaviour at the moment we would either need a separate collection for secret data that was locked down or write custom endpoints?

**Reply by ganigeorgiev** [reply]
@kisamoto It is only for specifying a subset of fields, aka. it doesn't have security purpose and it is intended to reduce the size of the transferred json response body.

> I assume that if we wanted this behaviour at the moment we would either need a separate collection for secret data that was locked down or write custom endpoints?

Yes, that's correct (_additionally, you also have the option to create a read-only [View collection](https://pocketbase.io/docs/collections/#view-collection)_).

**Comment by HEAVYPOLY** [comment]
<img width="667" alt="image" src="https://github.com/pocketbase/pocketbase/assets/24432677/0d0942b3-ac48-4d73-801b-9b0822118959">

I want everything in my user collection visible publicly except except subscriptionID, cusID.

Would I make the whole collection private, then create a read only public view collection for public info? 

Or is it safer to move subscriptionID and cusID to its own private collection and link them to users with relation?

**Reply by ganigeorgiev** [reply]
Both approaches are equally safe. It is up to you to decide how to structure your data based on how do you plan to manage those hidden fields.

For restricting access to sensitive data I personally would choose a new collection with a relation field to the main one.

**Reply by HEAVYPOLY** [reply]
great, thanks!
